### PR TITLE
gh-83004: Harden msvcrt init

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-83004

### CODE REVIEW

**Review by kumaraditya303 — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Darn, I forgot to add error handling for `PyDict_SetItemString`. I'll create a follow-up PR for that.

File: PC/msvcrtmodule.c [hunk 1]
```diff
@@ -644,30 +644,47 @@ PyInit_msvcrt(void)
...
-    if (st < 0) return NULL;
+    PyObject *version = PyUnicode_FromFormat("%d.%d.%d.%d",
+                                             _VC_CRT_MAJOR_VERSION,
+                                             _VC_CRT_MINOR_VERSION,
+                                             _VC_CRT_BUILD_VERSION,
+                                             _VC_CRT_RBUILD_VERSION);
+    if (version == NULL) {
+        goto error;
+    }
+    st = PyModule_AddObjectRef(m, "CRT_ASSEMBLY_VERSION", version);
```
Comment (zooba) [inline PC/msvcrtmodule.c:676]: @erlend-aasland Just FYI, the modsupport functions allow passing in `NULL` provided an error has already occurred, and they just let it chain. So the code here was fine before.

No need to revert now though.

Comment (erlend-aasland) [reply PC/msvcrtmodule.c:676]: TIL! Sorry for not waiting on your review.

Comment (erlend-aasland) [reply PC/msvcrtmodule.c:676]: However, there was a leak if PyModule_AddObject failed, since that API only eat refs if successful.

Comment (kumaraditya303) [reply PC/msvcrtmodule.c:676]: Sorry I should have been more careful in this.

**Pull request merged**